hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractContractCallTransactionHandler.java

```diff
@@ -89,6 +92,33 @@ protected final void onContractResult(RecordItem recordItem, ContractResult cont
 
             entityListener.onContractLog(contractLog);
         }
+
+        // contract call state changes
+        for (int index = 0; index < functionResult.getStateChangesCount(); ++index) {
+            com.hederahashgraph.api.proto.java.ContractStateChange contractStateChangeInfo = functionResult
+                    .getStateChanges(index);
+
+            var contractId = EntityId.of(contractStateChangeInfo.getContractID());
+            for (int i = 0; i < contractStateChangeInfo.getStorageChangesCount(); ++i) {
+                StorageChange storageChange = contractStateChangeInfo.getStorageChanges(index);
+
+                ContractStateChange contractStateChange = new ContractStateChange();
+                contractStateChange.setConsensusTimestamp(consensusTimestamp);
+                contractStateChange.setContractId(contractId);
+                contractStateChange.setPayerAccountId(contractResult.getPayerAccountId());
+                contractStateChange.setSlot(DomainUtils.toBytes(storageChange.getSlot()));
+                contractStateChange.setValueRead(DomainUtils.toBytes(storageChange.getValueRead()));
+
+                // If a value of zero is written the valueWritten will be present but the inner value will be absent.
+                // If a value was read and not written this value will not be present.
+                if (storageChange.isInitialized()) {
```

Comment:
I don't think this does what you want it to do. Regardless, it's more complex than it needs to be. The reason for using a wrapper object for `valueWritten` was to allow for a simple check.

```suggestion
                if (storageChange.hasValueWritten()) {
```

Reply:
Simplified